tripal/tripal
# OBO loader fix for double colons and an additional change to enhance load times for obos that require remote lookups

Issue #525 

## Type(s) of Change(s)

## Description
The obo loader was exploding entries on colons which caused entires like this: fatty acid 18:3, to load as this: fatty acid 18. This fix addresses that by limiting the number of exploded pieces. 

I also added additional steps to the obo loader when remote lookup is required. Remote look ups are now saved in the tripal_obo_temp table and before remote lookups are done that table is checked. This should significantly reduce load times as it should reduce the number of remote requests that need to be made during a single obo load.


## Testing?
To test:
1. In an environment where the plant trait ontology has not been loaded, checkout branch 525-tv3-obo_speed_colon, pull, and clear cache
2. Go to admin/tripal/loaders/chado_vocabs/obo_loader
3. Click on 'Add a New Ontology OBO Reference' to expand the form
4. Input the plant trait ontology using this url: http://purl.obolibrary.org/obo/to.obo
5. Click import OBO File
6. Run the drush command to run the job
7. Check the OBO loaded and in the cvterm table check that 'fatty acid 18:3' comes in as such and not 'fatty acid 18'


## Additional Notes (if any):
- [x ] My code follows the code style of this project.
